packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -190,16 +201,18 @@ export class RunningSummarizer implements IDisposable {
         }
     }
 
-    public handleOp(error: any, op: ISequencedDocumentMessage) {
+    public handleOp(error: any, { sequenceNumber, type, clientId, contents }: ISequencedDocumentMessage) {
         if (error !== undefined) {
             return;
         }
-        this.heuristicData.lastOpSequenceNumber = op.sequenceNumber;
+        this.heuristicData.lastOpSequenceNumber = sequenceNumber;
 
-        // Check for ops requesting summary
-        if (op.type === MessageType.Save) {
-            // TODO: cast is only required until TypeScript version 4.3
-            this.trySummarize(`save;${op.clientId}: ${op.contents}` as `save;${string}: ${string}`);
+        if (this.tryRunEnqueuedSummary()) {
+            // Intentionally do nothing; check for enqueued on-demand summaries
+        } else if (type === MessageType.Save) {
```

Comment:
If the message is save, shouldn't we summarize irrespective of whether we run an enqueued summary or not?
On a separate note, do we even use MessageType.Save?

Reply:
We don't use it as far as I know. We should just remove it at some point, but it's all the way up in protocol-definitions, and keeping this 2 lines of code is not a big deal for now. We could open an issue to deprecate it, unless someone wants it.

I wouldn't waste time fine-tuning the functionality. I treat MessageType.Save as an op to request a summary. So it has very loose guarantees. If an enqueued summary works out, then we don't need to do another one.

Reply:
I opened #6880 to remove

Reply:
I opened #6880 to remove